config: azurelinux 3.0 kernel config

allenpais opened the pull request:
Optimize and Converge Kernel Configuration for Enhanced Performance and Maintainability

This config is a based on:
https://github.com/microsoft/azurelinux/blob/3.0/SPECS/kernel/config

Key Changes:
  - Deprecated outdated configuration options that no longer provide benefits in current hardware and software contexts.

  - We are suggesting these optimizations for our WSL/LCOW convergence to azurelinux.

allenpais committed:
config: azurelinux 3.0 kernel config

Optimize and Converge Kernel Configuration for Enhanced
Performance and Maintainability

This config is a based on:
https://github.com/microsoft/azurelinux/blob/3.0/SPECS/kernel/config

Key Changes:
  - Deprecated outdated configuration options that no longer
    provide benefits in current hardware and software contexts.

  - We are suggesting these optimizations for our WSL/LCOW
    convergence to azurelinux.

Signed-off-by: Allen Pais <apais@linux.microsoft.com>

allenpais commented:
@rlmenge Please review.

tyhicks commented on SPECS/kernel/config:
```diff
@@ -293,11 +293,8 @@ CONFIG_TRACEPOINTS=y
-# CONFIG_KEXEC_SIG is not set
-# CONFIG_KEXEC_JUMP is not set
+# CONFIG_KEXEC is not set
+# CONFIG_KEXEC_FILE is not set
```
I don't think we should disable `CONFIG_KEXEC_FILE`. I would be fine with disabling `CONFIG_KEXEC`, however, since it cannot and will not ever support signature verification.

tyhicks replied:
@allenpais ping on this. `CONFIG_KEXEC_FILE` still needs to be enabled for kexec support. We only want to disable `CONFIG_KEXEC`.

tyhicks commented:
@allenpais I was hoping we could disable all but perhaps one of the `CONFIG_RD_*` options to reduce image size.

allenpais committed:
config: additional updates

 As suggested by Tyler:
   Keep CONFIG_KEXEC_FILE enabled.
   Remove unused CONFIG_RD_* configs.

Signed-off-by: Allen Pais <apais@linux.microsoft.com>

allenpais commented:
@tyhicks Thanks for taking the time out to review. I have addressed your feedback. There will probably be one or two more commits that will go in. I should have it pushed soon.

rlmenge reviewed (commented):
I was able to cherry-pick a majority of these changes onto a test branch and build a test azl3 RPM!
- created new [branch ](https://github.com/microsoft/azurelinux/commits/rlmenge/test/user/apais/azlinux-3.0-config/) off of [6.6.29.1 upgrade PR](https://github.com/microsoft/azurelinux/pull/8979)
- cherry-picked the 2 commits from this PR (required a small git merge to match 6.6.29.1) 
- updated the spec and configs to match Azure Linux toolchain for 3.0
- Built the rpm :) ([pipeline run](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=561341&view=artifacts&pathAsName=false&type=publishedArtifacts))

It will not properly make an initramfs at the moment because [dracut](https://github.com/microsoft/azurelinux/blob/3.0-dev/SPECS/dracut/defaults.conf#L6) explicitly expects xen modules. We can/should change dracut from relying on xen but this along with the kernel config xen changes should be its own PR

allenpais committed:
config: additional config changes

  - Re-enable support for XEN
  - Additional configs required to boot LCOW

Signed-off-by: Allen Pais <apais@linux.microsoft.com>

Camelron commented on SPECS/kernel/config:
```diff
@@ -1,24 +1,24 @@
+CONFIG_CC_VERSION_TEXT="gcc (GCC) 11.2.0"
 CONFIG_CC_IS_GCC=y
-CONFIG_GCC_VERSION=130200
+CONFIG_GCC_VERSION=110200
```
this is a detail of the toolchain you ran make on; we can safely revert this section to its original 3.0-dev values.

Camelron commented on SPECS/kernel/config:
```diff
@@ -1854,10 +1849,10 @@ CONFIG_MAC80211_LEDS=y
-CONFIG_NET_9P_XEN=m
+CONFIG_NET_9P=y
+CONFIG_NET_9P_FD=y
+CONFIG_NET_9P_VIRTIO=y
```
We are simultaneously trying to figure out the set of kernel modules to include in our initramfs. Dracut is rather pedantic about this; I'm pretty sure it'll complain if a module stops existing (even if it is built-in). 

https://github.com/microsoft/azurelinux/pull/9144/files#diff-dddc34dca2f7bebb34fd880ce85f99a512ca0e73d5e2ecf571b0b741ea4c07f2R2

Camelron commented:
> I was able to cherry-pick a majority of these changes onto a test branch and build a test azl3 RPM!
> 
> * created new [branch ](https://github.com/microsoft/azurelinux/commits/rlmenge/test/user/apais/azlinux-3.0-config/) off of [6.6.29.1 upgrade PR](https://github.com/microsoft/azurelinux/pull/8979)
> * cherry-picked the 2 commits from this PR (required a small git merge to match 6.6.29.1)
> * updated the spec and configs to match Azure Linux toolchain for 3.0
> * Built the rpm :) ([pipeline run](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=561341&view=artifacts&pathAsName=false&type=publishedArtifacts))
> 
> It will not properly make an initramfs at the moment because [dracut](https://github.com/microsoft/azurelinux/blob/3.0-dev/SPECS/dracut/defaults.conf#L6) explicitly expects xen modules. We can/should change dracut from relying on xen but this along with the kernel config xen changes should be its own PR

Right, we are currently in the middle of a rework to the set of kernel modules needed by dracut which we will have to sync with this change: https://github.com/microsoft/azurelinux/pull/9144

allenpais committed:
config: additional configs that are need by LCOW to boot

Signed-off-by: Allen Pais <apais@linux.microsoft.com>

Commit:
kernel: config: build in configs required for WSL2 boot

Build in configs that are called and used during the WSL2 boot.

NF/NFT/NETLINK configs are required for when mirrored networking is
enabled.

DXGKRNL creates /dev/dxg device, when vGPU is available. Driver is not
called by applications unless the /dev/dxg already exists.

Signed-off-by: Kelsey Steele <kelseysteele@microsoft.com>

allenpais committed:
Merge pull request #1 from kelsey-steele/user/kms/x86-kernel-config-updates

kernel: config: build in configs required for WSL2 boot

allenpais committed:
Revert "kernel: config: build in configs required for WSL2 boot"

This reverts commit f96b7b199f7ea5b43fabb28dedc167607bc68bf9.

 Revert WSL config changes.

Signed-off-by: Allen Pais <apais@linux.microsoft.com>
